File: leetcode/118.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> pascal = new ArrayList<>();

        for (int i = 0; i < numRows; i++) {
            Integer[] temp = new Integer[i+1];
            Arrays.fill(temp, 1);
            pascal.add(Arrays.asList(temp));
        }

        for (int i = 2; i < numRows; i++) {
            for (int j = 1; j < pascal.get(i).size() - 1; j++) {
                pascal.get(i).set(j, pascal.get(i-1).get(j-1) + pascal.get(i-1).get(j));
            }
        }
        return pascal;
    }
}

File: solutions/118.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        pascal = []
        for row in range(1, numRows + 1):
            temp = []
            res = 1
            temp.append(res)
            for col in range(1, row):
                res = res * (row - col) // col
                temp.append(res)
            pascal.append(temp)

        return pascal